build/types/knockout.d.ts

```diff
@@ -329,8 +329,19 @@ export const extenders: Extenders<any>;
 
 //#region subscribables/mappingHelpers.js
 
-type Unwrapped<T> = T extends ko.Subscribable<infer R> ? R :
-    T extends Record<any, any> ? { [P in keyof T]: Unwrapped<T[P]> } : T;
+type Unwrapped<T> = T extends ko.ObservableArray<infer R>
+    ? Unwrapped<R>[]
+    : T extends ko.Subscribable<infer R>
+    ? (
```

Comment:
`ko.toJS` actually recursively unwraps observables, so this could be `Unwrapped<R>`.

Reply:
Unfortunately typescript only allows recursive types to be used with array or prop mapping. This type is technically correct, but doesn't compile:

```typescript
export type Unwrapped<T> = T extends ko.Subscribable<infer R>
    ? (
        R extends Record<any, any>
            ? { [P in keyof R]: Unwrapped<R[P]>}
            : Unwrapped<R>
    )
    : T extends Date | RegExp | Function
    ? T
    : T extends Record<any, any>
    ? { [P in keyof T]: Unwrapped<T[P]> } 
    : T
```

Throws `Type alias 'Unwrapped' circularly references itself`

---

The best I'm able to come up with is adding another guard and bailing and assigning an explicit `unknown` type when recursive observables are encountered:

```typescript
export type Unwrapped<T> = T extends ko.Subscribable<infer R>
    ? (
        R extends Record<any, any>
            ? { [P in keyof R]: Unwrapped<R[P]>}
            : R extends ko.Subscribable
                ? unknown
                : R
    )
    : T extends Date | RegExp | Function
    ? T
    : T extends Record<any, any>
    ? { [P in keyof T]: Unwrapped<T[P]> } 
    : T
```

The difference between `any` and `unknown` is that consumers would not get silent failures, they would instead be forced to add a type notation before using the value.